### ISSUE TITLE
Introduce and enforce MSRV

### ISSUE THREAD

**Issue by tnull** [RCasatta/electrsd#56]
A dependency of `zip` recently bumped the `rustc` requirement to 1.6, leading to MSRV violations in downstream projects such as LDK (cf. https://github.com/lightningdevkit/rust-lightning/pull/2055) and BDK:

```
error: package `base64ct v1.6.0` cannot be built because it requires rustc 1.60 or newer, while the currently active rustc version is 1.57.0
```

It would be great if this crate and `bitcoind` could pin an old version of `zip` and enforce a reasonable MSRV, e.g., 1.48, which would allow us to keep using them.

Happy to open corresponding PRs after a concept ACK.

**Comment by RCasatta** [comment]
Pinning could break other projects, I would like to avoid it if possible. 

What do you think of the [approach](https://github.com/RCasatta/bitcoind/pull/102) followed by `bitcoind` (without features MSRV is 1.41.1) ?

Obviously, without the feature, you need to provide `bitcoind` executable from env var or PATH

**Comment by danielabrozzoni** [comment]
One thing we have in rust-hwi in order to avoid pinning dependencies for everyone (as it caused some troubles: https://github.com/bitcoindevkit/rust-hwi/issues/72) is a feature with a lower msrv, in which the dependencies are pinned: https://github.com/bitcoindevkit/rust-hwi/pull/73

I personally don't like the approach, but it seems to work

**Comment by RCasatta** [comment]
yeah, I think pinning via a specific feature is not great but the best option
